pkg/auth/providers/local/local_auth_provider.go

```diff
@@ -448,8 +449,10 @@ func minOf(length int, defaultLen int) int {
 func indexField(field string, maxIndex int) []string {
 	var fieldIndexes []string
 	for i := 2; i <= maxIndex; i++ {
-		fieldIndexes = append(fieldIndexes, strings.ToLower(
-			string([]rune(simplifyString(field))[0:i])))
+		fi := string([]rune(simplifyString(field))[0:i])
+		fieldIndexes = append(fieldIndexes, strings.TrimFunc(strings.ToLower(fi), func(r rune) bool {
+			return r == 0
```

Comment:
why do we need to remove the Unicode null character here?

Reply:
Great question...

So, `maxIndex` here is computed as `minOf(len(user.DisplayName), searchIndexDefaultLen)` (which I'll fix, because this can now be [`min()`](https://pkg.go.dev/builtin#min).

The slicing of this string (for the indexing) was going beyond the string length, and picking up 0x00 characters.

I reworked this to reslice on another `min(new length, maxIndex)` which is neater (I think).

Reply:
it's clearer now, thanks!